.github/workflows/run-e2e-tests.yaml

```diff
@@ -0,0 +1,59 @@
+#  This GitHub action is designed to run end-to-end (e2e) tests on a Kubernetes cluster using the k3s distribution.
+#  The action is triggered on pull requests to the main branch of the repository.
+#  The action consists of a single job that runs on an Ubuntu-latest virtual machine and has the necessary permissions to read and write various resources.
+#  The job has several steps:
+#    - Checkout the repository code.
+#    - Cache the ~/.cache/go-build and ~/go/pkg/mod directories to improve performance.
+#    - Set up a docker registry and k3s cluster using a script called k3s-setup.sh.
+#    - Set the pull request number, OCI module image name as environment variables.
+#    - Wait until the registry contains OCI module image for current PR (pushed by prow-job)
+#    - Run test script located in the scripts/testing directory
+
+name: Run E2E tests on the cluster
+
+env:
+  BTP_OPERATOR_REPO: europe-docker.pkg.dev/kyma-project/dev/unsigned/component-descriptors/kyma.project.io/module/btp-operator
+  BTP_MANAGER_REPO: europe-docker.pkg.dev/kyma-project/dev/btp-manager
+  MODULE_TAG_PREFIX: 0.0.0
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  run-e2e-tests:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+      id-token: write
+
+    steps:
+      - uses: actions/checkout@v3 # TODO could we narrow down what we check out (scripts would do)
+
+      - uses: actions/cache@v3 # TODO do we need cache here?
```

Comment:
yes, it makes it possible not to download the dependencies every time and overall speeds up the test quite bit

Reply:
Till, the reason I have some doubts is that if we check out only scripts (since e2e test are based on shell scripts for the time being) the cache would not be necessary. Unless we are using go test again.